### PR TITLE
Update docs about form types

### PR DESCRIPTION
**Pull request by kunicmarko20**
I am targeting this branch, because I am updating documentation about form types.

## Changelog

```markdown
### Changed
- Changed documentation for Form Types.
```

## Subject

After finishing my last PR and updating these docs about form types I saw that all other types need an update, so here it goes, the only thing I am not sure about is the CollectionTypes as that part is a bit confusing because of the same class name but a different bundle.

### CODE REVIEW

**Comment by greg0ire** [comment]
Can you check how it looks? I added something about how to generate the docs in contributing.md

**Comment by kunicmarko20** [comment]
Looks good to me, not sure about the backticks on the title:

screenshot from 2017-11-20 21-20-55

**Comment by greg0ire** [comment]
I think it looks good too, even the backtick in the title: people should now that it's a class. But now that I think of it, I think @jlamur asked me to only keep the `CollectionType` part on another similar PR

**Comment by kunicmarko20** [comment]
So I should remove the `sonata_type_native_collection` docs then?

File: docs/reference/form_types.rst [hunk 1]
```diff
@@ -4,18 +4,17 @@ Form Types
...
+When defining fields in your admin classes you can use any of the standard
 `Symfony field types`_ and configure them as you would normally. In addition
 there are some special Sonata field types which allow you to work with
 relationships between one entity class and another.
 
 .. _field-types-model:
 
-sonata_type_model
-^^^^^^^^^^^^^^^^^
+``Sonata\AdminBundle\Form\Type\ModelType``
```
Comment (greg0ire) [inline docs/reference/form_types.rst]: I think we elected to use just ` ``ModelType`` ` on another PR I made. @sonata-project/contributors , do you remember which repo?

**Comment by greg0ire** [comment]
No I meant something else, I added a better comment to explain it

File: docs/reference/form_types.rst (hunk 1 above)
Comment (kunicmarko20) [reply docs/reference/form_types.rst]: so just the class name and not the FQCN, but then there will be a problem with `Sonata\CoreBundle\Form\Type\CollectionType` and `Sonata\AdminBundle\Form\Type\CollectionType`

Comment (greg0ire) [reply docs/reference/form_types.rst]: 🤔 Hmmm right. We should probably keep it like this then...

**Comment by greg0ire** [comment]
Thanks a lot @kunicmarko20 !

**Pull request merged**